financial-connections/src/main/java/com/stripe/android/financialconnections/features/networkinglinkverification/NetworkingLinkVerificationViewModel.kt

```diff
@@ -62,41 +62,72 @@ internal class NetworkingLinkVerificationViewModel @AssistedInject constructor(
     private val logger: Logger,
     private val isLinkWithStripe: IsLinkWithStripe,
     private val attachConsumerToLinkAccountSession: AttachConsumerToLinkAccountSession,
+    private val getCachedConsumerSession: GetCachedConsumerSession,
 ) : FinancialConnectionsViewModel<NetworkingLinkVerificationState>(initialState, nativeAuthFlowCoordinator) {
 
     init {
         observeAsyncs()
         viewModelScope.launch {
             setState { copy(payload = Loading()) }
-            runCatching { getOrFetchSync().manifest.also { requireNotNull(it.accountholderCustomerEmailAddress) } }
-                .onSuccess { manifest ->
-                    lookupConsumerAndStartVerification(
-                        email = requireNotNull(manifest.accountholderCustomerEmailAddress),
-                        businessName = manifest.businessName,
-                        verificationType = VerificationType.SMS,
-                        onConsumerNotFound = {
-                            analyticsTracker.track(VerificationError(PANE, ConsumerNotFoundError))
-                            navigationManager.tryNavigateTo(InstitutionPicker(referrer = PANE))
-                        },
-                        onLookupError = { error ->
-                            analyticsTracker.track(VerificationError(PANE, LookupConsumerSession))
-                            setState { copy(payload = Fail(error)) }
-                        },
-                        onStartVerification = { /* no-op */ },
-                        onVerificationStarted = { consumerSession ->
-                            val payload = buildPayload(consumerSession, manifest)
-                            setState { copy(payload = Success(payload)) }
-                        },
-                        onStartVerificationError = { error ->
-                            analyticsTracker.track(
-                                VerificationError(PANE, StartVerificationSessionError)
-                            )
-                            setState { copy(payload = Fail(error)) }
-                        }
-                    )
-                }
-                .onFailure { setState { copy(payload = Fail(it)) } }
+
+            runCatching {
+                buildInitData()
+            }.onSuccess {
+                startVerification(it)
+            }.onFailure {
+                setState { copy(payload = Fail(it)) }
+            }
+        }
```

Comment:
I know it was not there before, but maybe we can replace this by a regular `suspend {...}.execute()`? If it's too much work we can defer it to the `improvements` backlog.

Reply:
Let’s discuss later this week. Together with [BANKCON-11754](https://jira.corp.stripe.com/browse/BANKCON-11754), we can rethink this more broadly.